static/js/containers/CourseSearchPage.js

```diff
@@ -84,6 +93,12 @@ export class CourseSearchPage extends React.Component<Props, State> {
     this.runSearch()
   }
 
+  componentDidUpdate(prevProps: Object, prevState: Object) {
+    if (shouldRunSearch(prevState, this.state)) {
+      this.runSearch()
+    }
+  }
```

Comment:
do you think we need to add a test for this little bit of logic?

Reply:
will add it

Reply:
There were already tests to verify a search was run if the text or a subject facet changed, I added another test to verify that no search is run if no search parameters are changed.